### PR TITLE
models: add function to check if quota is exceeded

### PR DESCRIPTION
**Pull request by mvidalgarcia**
closes reanahub/reana-server#282

### CODE REVIEW

File: reana_db/models.py [hunk 1]
```diff
@@ -236,6 +236,12 @@ def get_quota_usage(self):
             cpu=self.get_user_quota_by_type(ResourceType.cpu),
         )
 
+    def has_exceeded_quota(self):
```
Comment (mvidalgarcia) [inline reana_db/models.py:239]: I'm wondering if there should be a check per resource type as there might be some actions that the user can perform even when only the CPU quota is exceeded (create workflows, upload files, etc.) and the other way around (starting workflows).

Comment (mvidalgarcia) [reply reana_db/models.py:239]: Decided offline to remain conservative and prevent the mentioned user actions when the quota of **any** user resource is exceeded.

File: reana_db/models.py [hunk 2]
```diff
@@ -242,13 +242,19 @@ def initialize_user_quota_limits(self):
                 )
             )
 
+    def has_exceeded_quota(self):
+        """Get whether user has exceeded the quota of any resource."""
+        return any(
+            r.quota_limit != 0 and r.quota_used >= r.quota_limit for r in self.resources
+        )
+
+>>>>>>> models: add function to check if quota is exceeded
```
Comment (alintulu) [inline reana_db/models.py:251]: Possible left over from a merge conflict?

Comment (mvidalgarcia) [reply reana_db/models.py:251]: 🤦🏻‍♂️

**Review by diegodelemos — APPROVED** [review]

**Pull request merged**